# Prosper Loan Data Analysis

# Deepak Sithu

## Table of Contents

* <a href="#overview">Overview</a>
* <a href="#prelim">Preliminary Wrangling</a>
* <a href="#initial">Initial Exploration</a>
* <a href="#univar">Univariate Exploration</a>
* <a href="#bivar">Bivariate Exploration</a>
* <a href="#multivar">Multivariate Exploration</a>

In [1]:
# import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

<a id="overview"></a>
## Overview

> This project is an examination of loan data from Prosper. The data set contains information for 

> From the guide: This data set contains 113,937 loans with 81 variables on each loan, including loan amount, borrower rate (or interest rate), current loan status, borrower income, and many others. I'm supposed to focus on about 10-15 of the variables. Example questions are, what factors affect a loan's outcome status? What affects the borrower's APR or interest rate? Are there differences between loans depending on how large the original loan amount was?

In [2]:
# read in dataset
prosper = pd.read_csv('prosperLoanData.csv')

<a id="initial"></a>
## Initial Exploration

In [3]:
# get shape of dataframe and information for columns
print(prosper.shape)
prosper.info()

(113937, 81)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 113937 entries, 0 to 113936
Data columns (total 81 columns):
ListingKey                             113937 non-null object
ListingNumber                          113937 non-null int64
ListingCreationDate                    113937 non-null object
CreditGrade                            28953 non-null object
Term                                   113937 non-null int64
LoanStatus                             113937 non-null object
ClosedDate                             55089 non-null object
BorrowerAPR                            113912 non-null float64
BorrowerRate                           113937 non-null float64
LenderYield                            113937 non-null float64
EstimatedEffectiveYield                84853 non-null float64
EstimatedLoss                          84853 non-null float64
EstimatedReturn                        84853 non-null float64
ProsperRating (numeric)                84853 non-null float64
ProsperRa

In [4]:
# descriptions of numeric columns
prosper.describe()

ListingNumber           Term    BorrowerAPR   BorrowerRate  \
count   1.139370e+05  113937.000000  113912.000000  113937.000000   
mean    6.278857e+05      40.830248       0.218828       0.192764   
std     3.280762e+05      10.436212       0.080364       0.074818   
min     4.000000e+00      12.000000       0.006530       0.000000   
25%     4.009190e+05      36.000000       0.156290       0.134000   
50%     6.005540e+05      36.000000       0.209760       0.184000   
75%     8.926340e+05      36.000000       0.283810       0.250000   
max     1.255725e+06      60.000000       0.512290       0.497500   

         LenderYield  EstimatedEffectiveYield  EstimatedLoss  EstimatedReturn  \
count  113937.000000             84853.000000   84853.000000     84853.000000   
mean        0.182701                 0.168661       0.080306         0.096068   
std         0.074516                 0.068467       0.046764         0.030403   
min        -0.010000                -0.182700       0.004900        -0.182700   
25%         0.124200                 0.115670       0.042400         0.074080   
50%         0.173000                 0.161500       0.072400         0.091700   
75%         0.240000                 0.224300       0.112000         0.116600   
max         0.492500                 0.319900       0.366000         0.283700   

       ProsperRating (numeric)  ProsperScore  ...  LP_ServiceFees  \
count             84853.000000  84853.000000  ...   113937.000000   
mean                  4.072243      5.950067  ...      -54.725641   
std                   1.673227      2.376501  ...       60.675425   
min                   1.000000      1.000000  ...     -664.870000   
25%                   3.000000      4.000000  ...      -73.180000   
50%                   4.000000      6.000000  ...      -34.440000   
75%                   5.000000      8.000000  ...      -13.920000   
max                   7.000000     11.000000  ...       32.060000   

       LP_CollectionFees  LP_GrossPrincipalLoss  LP_NetPrincipalLoss  \
count      113937.000000          113937.000000        113937.000000   
mean          -14.242698             700.446342           681.420499   
std           109.232758            2388.513831          2357.167068   
min         -9274.750000             -94.200000          -954.550000   
25%             0.000000               0.000000             0.000000   
50%             0.000000               0.000000             0.000000   
75%             0.000000               0.000000             0.000000   
max             0.000000           25000.000000         25000.000000   

       LP_NonPrincipalRecoverypayments  PercentFunded  Recommendations  \
count                    113937.000000  113937.000000    113937.000000   
mean                         25.142686       0.998584         0.048027   
std                         275.657937       0.017919         0.332353   
min                           0.000000       0.700000         0.000000   
25%                           0.000000       1.000000         0.000000   
50%                           0.000000       1.000000         0.000000   
75%                           0.000000       1.000000         0.000000   
max                       21117.900000       1.012500        39.000000   

       InvestmentFromFriendsCount  InvestmentFromFriendsAmount      Investors  
count               113937.000000                113937.000000  113937.000000  
mean                     0.023460                    16.550751      80.475228  
std                      0.232412                   294.545422     103.239020  
min                      0.000000                     0.000000       1.000000  
25%                      0.000000                     0.000000       2.000000  
50%                      0.000000                     0.000000      44.000000  
75%                      0.000000                     0.000000     115.000000  
max                     33.000000                 25000.000000    1189.000000

In [5]:
# head of dataframe
prosper.head()

ListingKey  ListingNumber            ListingCreationDate  \
0  1021339766868145413AB3B         193129  2007-08-26 19:09:29.263000000   
1  10273602499503308B223C1        1209647  2014-02-27 08:28:07.900000000   
2  0EE9337825851032864889A          81716  2007-01-05 15:00:47.090000000   
3  0EF5356002482715299901A         658116  2012-10-22 11:02:35.010000000   
4  0F023589499656230C5E3E2         909464  2013-09-14 18:38:39.097000000   

  CreditGrade  Term LoanStatus           ClosedDate  BorrowerAPR  \
0           C    36  Completed  2009-08-14 00:00:00      0.16516   
1         NaN    36    Current                  NaN      0.12016   
2          HR    36  Completed  2009-12-17 00:00:00      0.28269   
3         NaN    36    Current                  NaN      0.12528   
4         NaN    36    Current                  NaN      0.24614   

   BorrowerRate  LenderYield  ...  LP_ServiceFees  LP_CollectionFees  \
0        0.1580       0.1380  ...         -133.18                0.0   
1        0.0920       0.0820  ...            0.00                0.0   
2        0.2750       0.2400  ...          -24.20                0.0   
3        0.0974       0.0874  ...         -108.01                0.0   
4        0.2085       0.1985  ...          -60.27                0.0   

   LP_GrossPrincipalLoss  LP_NetPrincipalLoss LP_NonPrincipalRecoverypayments  \
0                    0.0                  0.0                             0.0   
1                    0.0                  0.0                             0.0   
2                    0.0                  0.0                             0.0   
3                    0.0                  0.0                             0.0   
4                    0.0                  0.0                             0.0   

   PercentFunded  Recommendations InvestmentFromFriendsCount  \
0            1.0                0                          0   
1            1.0                0                          0   
2            1.0                0                          0   
3            1.0                0                          0   
4            1.0                0                          0   

  InvestmentFromFriendsAmount Investors  
0                         0.0       258  
1                         0.0         1  
2                         0.0        41  
3                         0.0       158  
4                         0.0        20  

[5 rows x 81 columns]

### What is the structure of the dataset?

> The dataset is a series of loans with a large amount of associated data for each. One interesting aspect is that the credit grading system changed in July 2009, so credit ratings are split between the columns CreditGrade and ProsperRating (Alpha). It's clear from the initial exploration into the dataset that, both for ease of use and for the scope of this project it is best to eliminate several of the columns which are not going to be relavent. 

### What are the main features of interest in the dataset?

> The main features of interest are the credit grades, which are split between CreditGrade and ProsperRating (Alpha), the LoanStatus, the BorrowerAPR, and the BorrowerRate. It will be interesting to see what, if any of the other variables in the dataset are correlated with differences in the previously mentioned features. Many of these are features which are  determined by the loan company, and therefore out of the control of the borrower, so seeing the relationship between these and variables which are more borrower-based will provide an insight into what goes into determining the specifics of a loan.

### What features in the dataset do you think will help support your investigation into the features of interest?

> The features which will be explored to see if they support an investigation into the features of interest are Term, ListingCategory, BorrowerState, EmploymentStatus, IsBorrowerHomeowner, OpenCreditLines, TotalCreditLinespast7years, DebtToIncomeRatio, and IncomeRange.

<a id="prelim"></a>
## Data Cleaning

Issue
Define
Code
Test

#### Issue: There are many unnecessary columns in the dataset which are outside of the scope of this project.

In [111]:
# create a copy of the dataframe with only columns the of interest and confirm change
prosper_clean = prosper[['ListingKey', 'CreditGrade', 'Term', 'LoanStatus', 'BorrowerAPR', 'BorrowerRate', 'ProsperRating (Alpha)', 'ListingCategory (numeric)', 'BorrowerState', 'EmploymentStatus', 'IsBorrowerHomeowner', 'OpenCreditLines', 'TotalCreditLinespast7years', 'DebtToIncomeRatio', 'IncomeRange']].copy()
print(prosper_clean.shape)
prosper_clean.head()

(113937, 15)


ListingKey CreditGrade  Term LoanStatus  BorrowerAPR  \
0  1021339766868145413AB3B           C    36  Completed      0.16516   
1  10273602499503308B223C1         NaN    36    Current      0.12016   
2  0EE9337825851032864889A          HR    36  Completed      0.28269   
3  0EF5356002482715299901A         NaN    36    Current      0.12528   
4  0F023589499656230C5E3E2         NaN    36    Current      0.24614   

   BorrowerRate ProsperRating (Alpha)  ListingCategory (numeric)  \
0        0.1580                   NaN                          0   
1        0.0920                     A                          2   
2        0.2750                   NaN                          0   
3        0.0974                     A                         16   
4        0.2085                     D                          2   

  BorrowerState EmploymentStatus  IsBorrowerHomeowner  OpenCreditLines  \
0            CO    Self-employed                 True              4.0   
1            CO         Employed                False             14.0   
2            GA    Not available                False              NaN   
3            GA         Employed                 True              5.0   
4            MN         Employed                 True             19.0   

   TotalCreditLinespast7years  DebtToIncomeRatio     IncomeRange  
0                        12.0               0.17  $25,000-49,999  
1                        29.0               0.18  $50,000-74,999  
2                         3.0               0.06   Not displayed  
3                        29.0               0.15  $25,000-49,999  
4                        49.0               0.26       $100,000+

#### Issue: the column name 'ProsperRating (Alpha)' affects ease of use of the dataset

In [112]:
# rename ProsperRating (Alpha) and confirm change
prosper_clean = prosper_clean.rename(columns={'ProsperRating (Alpha)' : 'ProsperRating'});
prosper_clean.head()

ListingKey CreditGrade  Term LoanStatus  BorrowerAPR  \
0  1021339766868145413AB3B           C    36  Completed      0.16516   
1  10273602499503308B223C1         NaN    36    Current      0.12016   
2  0EE9337825851032864889A          HR    36  Completed      0.28269   
3  0EF5356002482715299901A         NaN    36    Current      0.12528   
4  0F023589499656230C5E3E2         NaN    36    Current      0.24614   

   BorrowerRate ProsperRating  ListingCategory (numeric) BorrowerState  \
0        0.1580           NaN                          0            CO   
1        0.0920             A                          2            CO   
2        0.2750           NaN                          0            GA   
3        0.0974             A                         16            GA   
4        0.2085             D                          2            MN   

  EmploymentStatus  IsBorrowerHomeowner  OpenCreditLines  \
0    Self-employed                 True              4.0   
1         Employed                False             14.0   
2    Not available                False              NaN   
3         Employed                 True              5.0   
4         Employed                 True             19.0   

   TotalCreditLinespast7years  DebtToIncomeRatio     IncomeRange  
0                        12.0               0.17  $25,000-49,999  
1                        29.0               0.18  $50,000-74,999  
2                         3.0               0.06   Not displayed  
3                        29.0               0.15  $25,000-49,999  
4                        49.0               0.26       $100,000+

#### Issue: ListingCategory (numeric) only has numeric values representing each listing type, but it would be more meaningful to include the actual listing types.

In [113]:
# create empty column ListingCategory to replace ListingCategory (numeric)
prosper_clean['ListingCategory'] = ""

In [114]:
# loop through dataset to fill ListingCategory based on ListingCategory (numeric)
for i in range(len(prosper_clean)):
    num = prosper_clean['ListingCategory (numeric)'][i]
    if num == 0:
        prosper_clean.at[i, 'ListingCategory'] = 'Not Available'
    elif num == 1:
        prosper_clean.at[i, 'ListingCategory'] = 'Debt Consolidation'
    elif num == 2:
        prosper_clean.at[i, 'ListingCategory'] = 'Home Improvement'
    elif num == 3:
        prosper_clean.at[i, 'ListingCategory'] = 'Business'
    elif num == 4:
        prosper_clean.at[i, 'ListingCategory'] = 'Personal Loan'
    elif num == 5:
        prosper_clean.at[i, 'ListingCategory'] = 'Student Use'
    elif num == 6:
        prosper_clean.at[i, 'ListingCategory'] = 'Auto'
    elif num == 7:
        prosper_clean.at[i, 'ListingCategory'] = 'Other'
    elif num == 8:
        prosper_clean.at[i, 'ListingCategory'] = 'Baby/Adoption'
    elif num == 9:
        prosper_clean.at[i, 'ListingCategory'] = 'Boat'
    elif num == 10:
        prosper_clean.at[i, 'ListingCategory'] = 'Cosmetic Procedure'
    elif num == 11:
        prosper_clean.at[i, 'ListingCategory'] = 'Engagement Ring'
    elif num == 12:
        prosper_clean.at[i, 'ListingCategory'] = 'Green Loans'
    elif num == 13:
        prosper_clean.at[i, 'ListingCategory'] = 'Household Expenses'
    elif num == 14:
        prosper_clean.at[i, 'ListingCategory'] = 'Large Purchases'
    elif num == 15:
        prosper_clean.at[i, 'ListingCategory'] = 'Medical/Dental'
    elif num == 16:
        prosper_clean.at[i, 'ListingCategory'] = 'Motorcycle'
    elif num == 17:
        prosper_clean.at[i, 'ListingCategory'] = 'RV'
    elif num == 18:
        prosper_clean.at[i, 'ListingCategory'] = 'Taxes'
    elif num == 19:
        prosper_clean.at[i, 'ListingCategory'] = 'Vacation'
    else:
        prosper_clean.at[i, 'ListingCategory'] = 'Wedding Loans'

In [115]:
# remove ListingCategory (numeric) from clean dataframe and confirm changes
prosper_clean = prosper_clean.drop('ListingCategory (numeric)', axis = 1)
prosper_clean.head()

ListingKey CreditGrade  Term LoanStatus  BorrowerAPR  \
0  1021339766868145413AB3B           C    36  Completed      0.16516   
1  10273602499503308B223C1         NaN    36    Current      0.12016   
2  0EE9337825851032864889A          HR    36  Completed      0.28269   
3  0EF5356002482715299901A         NaN    36    Current      0.12528   
4  0F023589499656230C5E3E2         NaN    36    Current      0.24614   

   BorrowerRate ProsperRating BorrowerState EmploymentStatus  \
0        0.1580           NaN            CO    Self-employed   
1        0.0920             A            CO         Employed   
2        0.2750           NaN            GA    Not available   
3        0.0974             A            GA         Employed   
4        0.2085             D            MN         Employed   

   IsBorrowerHomeowner  OpenCreditLines  TotalCreditLinespast7years  \
0                 True              4.0                        12.0   
1                False             14.0                        29.0   
2                False              NaN                         3.0   
3                 True              5.0                        29.0   
4                 True             19.0                        49.0   

   DebtToIncomeRatio     IncomeRange   ListingCategory  
0               0.17  $25,000-49,999     Not Available  
1               0.18  $50,000-74,999  Home Improvement  
2               0.06   Not displayed     Not Available  
3               0.15  $25,000-49,999        Motorcycle  
4               0.26       $100,000+  Home Improvement

#### Issue: The ordered categorical variables CreditGrade, ProsperRating, and IncomeRange are not formatted as categorical type variables.

In [116]:
# turn CreditGrade, ProsperRating, and IncomeRange into ordered categorical type variables and confirm change
ordinal_var_dict = {'CreditGrade': ['NC', 'HR', 'E', 'D', 'C', 'B', 'A', 'AA'], 
                   'ProsperRating': ['HR', 'E', 'D', 'C', 'B', 'A', 'AA'],
                   'IncomeRange': ['Not displayed', 'Not employed', '$0', '$1-24,999', '$25,000-49,999', '$50,000-74,999', '$75,000-99,999', '$100,000+']}

for var in ordinal_var_dict:
    ordered_var = pd.api.types.CategoricalDtype(ordered = True, 
                                               categories = ordinal_var_dict[var])
    prosper_clean[var] = prosper_clean[var].astype(ordered_var)
    
prosper_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 113937 entries, 0 to 113936
Data columns (total 15 columns):
ListingKey                    113937 non-null object
CreditGrade                   28953 non-null category
Term                          113937 non-null int64
LoanStatus                    113937 non-null object
BorrowerAPR                   113912 non-null float64
BorrowerRate                  113937 non-null float64
ProsperRating                 84853 non-null category
BorrowerState                 108422 non-null object
EmploymentStatus              111682 non-null object
IsBorrowerHomeowner           113937 non-null bool
OpenCreditLines               106333 non-null float64
TotalCreditLinespast7years    113240 non-null float64
DebtToIncomeRatio             105383 non-null float64
IncomeRange                   113937 non-null category
ListingCategory               113937 non-null object
dtypes: bool(1), category(3), float64(5), int64(1), object(5)
memory usage: 10.0+ MB


#### Issue: The credit rating is split between the columns CreditGrade and ProsperRating, but would be useful as a single column.

In [117]:
# create two lists of the ListingKeys for non-null values for CreditGrade and ProsperRating
credit_grade_keys = prosper_clean.ListingKey[prosper_clean['CreditGrade'].notnull()].tolist();
prosper_rating_keys = prosper_clean.ListingKey[prosper_clean['ProsperRating'].notnull()].tolist();

In [118]:
# use python in and loop to check if any value in credit_grade_list is in prosper_rating_list
double_rating_keys = []
for i in range(len(credit_grade_keys)):
    if credit_grade_keys[i] in prosper_rating_keys:
        double_rating_keys.append(credit_grade_keys[i])
        print(credit_grade_keys[i])

In [119]:
# confirm double_rating_keys is empty
double_rating_keys

[]

In [120]:
# create empty CreditRating column to be filled by CreditGrade and ProsperRating
prosper_clean['CreditRating'] = "" 

In [121]:
# loop to fill CreditRating for CreditGrade values
for i in range(len(credit_grade_keys)):
    listing_row = (prosper_clean[prosper_clean['ListingKey'] == credit_grade_keys[i]])
    index = int(listing_row.index.values)
    prosper_clean.at[index, 'CreditRating'] = listing_row.CreditGrade.values[0]

In [122]:
# loop to fill CreditRating for ProsperRating values
for i in range(len(prosper_rating_keys)):
    listing_row = (prosper_clean[prosper_clean['ListingKey'] == prosper_rating_keys[i]])
    index = int(listing_row.index.values[0:1])
    prosper_clean.at[index, 'CreditRating'] = listing_row.ProsperRating.values[0]

In [143]:
# fill in empty CreditRatings as NC and confirm change
prosper_clean.CreditRating = prosper_clean.CreditRating.replace('', 'NC');
prosper_clean.head(25)

ListingKey CreditGrade  Term            LoanStatus  \
0   1021339766868145413AB3B           C    36             Completed   
1   10273602499503308B223C1         NaN    36               Current   
2   0EE9337825851032864889A          HR    36             Completed   
3   0EF5356002482715299901A         NaN    36               Current   
4   0F023589499656230C5E3E2         NaN    36               Current   
5   0F05359734824199381F61D         NaN    60               Current   
6   0F0A3576754255009D63151         NaN    36               Current   
7   0F1035772717087366F9EA7         NaN    36               Current   
8   0F043596202561788EA13D5         NaN    36               Current   
9   0F043596202561788EA13D5         NaN    36               Current   
10  0F123545674891886D9F106         NaN    60               Current   
11  0F1734025150298088A5F2B           C    36             Completed   
12  0F1A3597143888805163EF7         NaN    36  Past Due (1-15 days)   
13  0F1C3583260311305D68F87         NaN    36               Current   
14  0F353575943675863D1AFC0         NaN    60               Current   
15  0F483544120452347F48121         NaN    36             Defaulted   
16  0F4835843081468212F0892         NaN    60               Current   
17  0FDF336575182398852A8DC          AA    36            Chargedoff   
18  0FF2357105361877390B58D         NaN    36               Current   
19  0FFC35866018516621B0D3F         NaN    60               Current   
20  101635959323992333B3C70         NaN    36               Current   
21  1017340630713945364DA6B           D    36             Completed   
22  101735700321773921526BF         NaN    36               Current   
23  101B35764008108187B2EBA         NaN    36             Defaulted   
24  101B3596709094692AC79C0         NaN    36               Current   

    BorrowerAPR  BorrowerRate ProsperRating BorrowerState EmploymentStatus  \
0       0.16516        0.1580           NaN            CO    Self-employed   
1       0.12016        0.0920             A            CO         Employed   
2       0.28269        0.2750           NaN            GA    Not available   
3       0.12528        0.0974             A            GA         Employed   
4       0.24614        0.2085             D            MN         Employed   
5       0.15425        0.1314             B            NM         Employed   
6       0.31032        0.2712             E            KS         Employed   
7       0.23939        0.2019             C            CA         Employed   
8       0.07620        0.0629            AA            IL         Employed   
9       0.07620        0.0629            AA            IL         Employed   
10      0.27462        0.2489             C            MD         Employed   
11      0.15033        0.1325           NaN           NaN        Full-time   
12      0.17969        0.1435             B            AL         Employed   
13      0.13138        0.1034             A            AZ         Employed   
14      0.11695        0.0949             A            VA         Employed   
15      0.35797        0.3177            HR            FL            Other   
16      0.30748        0.2809             E            CA        Full-time   
17      0.13202        0.1250           NaN           NaN    Not available   
18      0.12528        0.0974             A            PA         Employed   
19      0.24754        0.2225             D            OR         Employed   
20      0.16732        0.1314             B            MN         Employed   
21      0.21488        0.2075           NaN            MI        Full-time   
22      0.35356        0.3134            HR            NY            Other   
23      0.28032        0.2419             D            IL         Employed   
24      0.19859        0.1620             C            LA         Employed   

    IsBorrowerHomeowner  OpenCreditLines  TotalCreditLinespast7years  \
0                  True              4.0                        12.0 

In [144]:
# convert CreditRating to ordered categorical type and confirm change
ordinal_var_dict = {'CreditRating': ['NC', 'HR', 'E', 'D', 'C', 'B', 'A', 'AA']}

for var in ordinal_var_dict:
    ordered_var = pd.api.types.CategoricalDtype(ordered = True, 
                                               categories = ordinal_var_dict[var])
    prosper_clean[var] = prosper_clean[var].astype(ordered_var)
    
prosper_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 113937 entries, 0 to 113936
Data columns (total 16 columns):
ListingKey                    113937 non-null object
CreditGrade                   28953 non-null category
Term                          113937 non-null int64
LoanStatus                    113937 non-null object
BorrowerAPR                   113912 non-null float64
BorrowerRate                  113937 non-null float64
ProsperRating                 84853 non-null category
BorrowerState                 108422 non-null object
EmploymentStatus              111682 non-null object
IsBorrowerHomeowner           113937 non-null bool
OpenCreditLines               106333 non-null float64
TotalCreditLinespast7years    113240 non-null float64
DebtToIncomeRatio             105383 non-null float64
IncomeRange                   113937 non-null category
ListingCategory               113937 non-null object
CreditRating                  113937 non-null category
dtypes: bool(1), category(4), flo

In [142]:
prosper

C     23745
B     19757
D     19323
A     17705
E     13005
HR    10425
AA     8834
NC     1143
Name: CreditRating, dtype: int64

In [110]:
prosper_clean

Remaining variables and meanings:
a1. ListingKey - ID, not for analysis - full
a2. CreditGrade - categorical (8) - 28953
a3. Term - categorical (3 (12, 36, 60 aka 1, 3, 5 year loans)) - full
4. LoanStatus - categorical (12) - full
5. BorrowerAPR - numeric - nearly full
6. BorrowerRate - numeric - full
7. ProsperRating (Alpha) - categorical (7) (CreditGrade for loans after July 2009) - 84853
8. ListingCategory (numeric) - categorical but need to translate it from numeric (20 categories) - full
9. BorrowerState - categorical (51) - some empty but mostly full
10. EmploymentStatus - categorical (8) - some emp but most full
11. IsBorrowerHomeowner - boolean - full
12. OpenCreditLines - numeric - some emp but mos full
13. TotalCreditLinespast7years - numeric - sebmf
14. DebtToIncomeRatio - numeric - sebmf
15. IncomeRange - categorical (8) - full

<a id="univar"></a>
## Univariate Exploration

In [ ]:
# Here explore distributions of individual variables,
# if you see unusual points or outliers, take a deeper look to clean things up 
# and prepare yourself to look at relationships between variables

In [ ]:
# Make sure that, after every plot or related series of plots,
#you include a markdown cell with comments about what you observed and what you plan on investigating next!!!

### Discuss the distributions of your variables of interest. Were there any unusual points? Did you need to perform any transformations?

> asdf

### Of the features you investigated, were there any unusual distributions? Did you perform any operations on the data to tidy, adjust, or change the form of the data? If so, why did you do this?

> asdf

<a id="bivar"></a>
## Bivariate Exploration

In [ ]:
# In this section, investigate relationships between pairs of variables in your data
# Make sure the variables that you cover here have been introduced in some fashion in the previous section (univariate)

### Talk about some of the relationships you observed in this part of the investigation. How did the features of interest vary with other features in the dataset?

> asdf

### Did you observe any interesting relationships between the other features (not the main features of interest)?

> asdf

<a id="multivar"></a>
## Multivariate Exploration

In [ ]:
# create plots of three or more variables to investigate your data further.
# make sure your investigations are justified, and follow from your work in the previous sections.

### Talk about some of the relationships you observed in this part of the investigation. Were there features that strengthened each other in terms of looking at your features of interest?

> asdf

### Were there any interesting or surprising interactions between features?

> asdf

> At the end of your report, make sure that you export the notebook as an
html file from the `File > Download as... > HTML` menu. Make sure you keep
track of where the exported file goes, so you can put it in the same folder
as this notebook for project submission. Also, make sure you remove all of
the quote-formatted guide notes like this one before you finish your report! XXXXXXX

there was an error in creating the CreditRatings column where certain values from the ProsperRatings column couldn't be indexed properly and so were put into CreditRatings as ''. This occured at most about 1,000 times, and likely fewer as there are some rows which had neither a CreditGrade nor Prosper Rating.